### PR TITLE
Superficial support for `Core.BigInt` type

### PR DESCRIPTION
**Pull request by zygoloid**
Add a `Core.BigInt` type and a corresponding builtin type in the toolchain. See [corresponding section of the design](https://docs.carbon-lang.dev/docs/design/expressions/literals.html#defined-types).

So far this type is not used for anything, and there is no way to create an instance of it.

### CODE REVIEW

**Comment by jonmeow** [comment]
Can you note in the PR description why you're doing this? I assume it's support for something, but it's non-obvious to me.

File: core/prelude/types.carbon [hunk 1]
```diff
@@ -11,6 +11,7 @@ export import library "prelude/types/bool";
 // noise in the toolchain tests.
 // import library "prelude/types/i32";
 
+fn BigInt() -> type = "bigint.make_type";
```
Comment (jonmeow) [inline core/prelude/types.carbon]:
```suggestion
fn BigInt() -> type = "big_int.make_type";
```

Suggesting for the underscore between words.

**Comment by zygoloid** [comment]
> Can you note in the PR description why you're doing this? I assume it's support for something, but it's non-obvious to me.

Added a link to the design. Did you want more than that? (Is your question why at all, or why now?)

**Review by jonmeow — APPROVED** [review]
LGTM, though would appreciate more information in the PR description before merging. [ed: I see this is now addressed]

File: core/prelude/types.carbon (hunk 1 above)
Comment (zygoloid) [reply core/prelude/types.carbon]: Done throughout.

**Pull request merged**